src/js/components/heading/Heading.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import { compose } from 'recompose';
+
+import StyledHeading from './StyledHeading';
+
+import { withTheme } from '../hocs';
+
+import doc from './doc';
+
+const styledComponents = {
+  div: StyledHeading,
+}; // tag -> styled component
+
+class Heading extends Component {
+  static defaultProps = {
+    level: 1,
+  };
+
+  render() {
+    const {
+      level,
+      ...rest
+    } = this.props;
+
+    const tag = `h${level}`;
+    let StyledComponent = styledComponents[tag];
+    if (!StyledComponent) {
+      StyledComponent = StyledHeading.withComponent(tag);
```

Comment:
for accessibility reasons I propose we don't allow changing the tag here.

Users with disabilities use tools to see all the heading in the page. and if we are not using the h1-h6 range they will not be able to see it.